packages/kit/src/runtime/client/client.js

```diff
@@ -372,13 +377,40 @@ export function create_client(app, target) {
 		load_cache = null;
 
 		if (started) {
+			const navigation = /** @type {import('types').OnNavigate} */ (
+				create_navigation(
+					current,
+					intent,
+					url,
+					/** @type {import('types').NavigationType} */ (type)
+				)
+			);
+
 			current = navigation_result.state;
 
 			// reset url before updating page store
 			if (navigation_result.props.page) {
 				navigation_result.props.page.url = url;
 			}
 
+			const after_navigate = (
+				await Promise.all(callbacks.on_navigate.map((fn) => fn(navigation)))
```

Comment:
should we consider what happens here if any of these callbacks reject? Or at least document the behavior? I'm not sure if it's the right behavior, but right now a rejected promise will prevent the navigation from completing.